### DISCUSSION TITLE
Thoughts on using automation tools to keep Transmission updated, scrutinized, and modernized?

### DISCUSSION THREAD

**Discussion by Pentaphon** [transmission/transmission#6557] [General]
It's been nearly 20 years since Transmission debuted and since we're just reaching the age of automation, machine learning, etc being available to the general public, I think it's worth having a discussion about what tools can be used to help a volunteer project like Transmission moving along. 

I think the project could use things like https://github.com/dependabot to keep the dependencies indexed and up-to-date. 

Anything else that could help our torrent client be the best it can be?

**Comment by GaryElshaw** [comment]
Here's the Transmission third-party directory https://github.com/transmission/transmission/tree/main/third-party

I'm not really looking at stuff that hasn't been updated for numerous reasons, for example libevent.

- dht and googletest haven't been updated for 2 years
- Libnatpmp 8 years
- miniupnpc 7 years
- Libpsl 3 years
- Libdeflate 9 months
- wide-integer 2 years
- utfcpp 3 years

It's a disservice not looking at release versions, but it's also worth looking at pre-release. I know some of the third-party dependencies that are included use commit changes rather than release versions. I would literally pr these changes if i had a better idea of how to do this. If anyone wants to talk me through how to do this, i'm all ears.
